File: 2021/2/task1.go
package main

import (
    "fmt"
    "os"
    "io/ioutil"
    s "strings"
    "strconv"
)


func getFileAsList(fn string) []string {
    data, err := ioutil.ReadFile(fn)
    if err != nil {
        fmt.Println("Error opening file")
        os.Exit(1)
    }
    str := s.Trim(s.Replace(string(data), "\r", "", -1), "\n")
    list := s.Split(str, "\n")
    return list
}


func Atoi(str string) int {
    val, err := strconv.Atoi(str)
    if err != nil {
        fmt.Println("Error converting string to number: ", str)
        os.Exit(2)
    }
    return val
}


func main() {
    list := getFileAsList("input.txt")
    depth := 0
    hpos := 0
    for _, line := range list {
        arr := s.Split(line, " ")
        cmd := arr[0]
        valstr := arr[1]
        val := Atoi(valstr)
        if (cmd == "forward") { hpos += val }
        if (cmd == "up")      { depth -= val }
        if (cmd == "down")    { depth += val }
    }
    fmt.Println("Depth:", depth, ", H-Pos: ", hpos, ", Product: ", depth*hpos)
}

File: 2021/6/task1.py
import numpy as np

lines = [x.rstrip() for x in open("input.txt")]
input = [int(x) for x in lines[0].split(",")]
print(input)

for i in range(256):
    numnew = 0
    for x in range(len(input)):
        if input[x] == 0:
            input[x] = 6
            numnew += 1
        else:
            input[x] -= 1
    for x in range(numnew):
        input.append(8)
print(len(input))
